src/sage/graphs/graph_decompositions/modular_decomposition.pyx

```diff
@@ -1410,3 +1464,31 @@ def recreate_decomposition(trials, algorithm, max_depth, max_fan_out,
         assert equivalent_trees(rand_tree, reconstruction)
         if verbose:
             print("Passes!")
+
+
+@random_testing
+def check_algos_are_equivalent(trials, graph_gen, verbose=False):
+    r"""
+    Verify that both algorithms compute the same tree (up to equivalence) for
+    random graphs.
+
+    EXAMPLES::
+
+        sage: from sage.graphs.graph_decompositions.modular_decomposition import *
+        sage: check_algos_are_equivalent(3, lambda : graphs.RandomGNP(10, 0.5))
```

Comment:
you could also add a test for a sparser graph, that is more likely to be disconnected.

Reply:
done in [451b325](https://github.com/sagemath/sage/pull/39038/commits/451b3256ac2cbbb00d2ff24ecc80339fe73d5e40)